docs/src/dev-manual/db-migrations.md

```diff
@@ -0,0 +1,205 @@
+`internal/persistence/ent/schema/` directory, then regenerate the [Ent]
+schema:
+
+```bash
+make gen-ent
+```
+
+### Generate the "enduro" migration file
+
+```bash
+go run ./cmd/migrate/ \
+--db="ingest" \
+--dsn="mysql://root:root123@tcp(localhost:3306)/enduro_migrate" \
+--path="./internal/db/migrations" \
+--name="add_sip_note_column"
+```
+
+The `--name` option should summarize the changes made to the schema. Have a look
+at the migration files in the `internal/db/migrations` directory for examples of
+migration names.
+
+If you need to change the name of a migration file you can manually change the
+filename then regenerate the [Atlas] hash file (see below), or you can revert all
+the changes to your local copy of Enduro and start again.
+
+You may also need to make changes to the generated migration file to add or
+modify the database data (e.g. add a default value for a column, map an existing
+value to a new value). If you do make changes to the migration file, always make
+sure to regenerate the [Atlas] hashes or the migration will fail.
+
+### Add ".up" to the "enduro" migration file extension
+
+```bash
+mv ./internal/db/migrations/20250423213837_add_sip_note_column.sql \
+./internal/db/migrations/20250423213837_add_sip_note_column.up.sql
+```
+
+Also change the file name in the `internal/db/migrations/atlas.sum`
+file.
+
+```text
+20250423213837_add_sip_note_column.up.sql h1:EM36pE8Uo0wK/Uzz8ZIEP9/p7D/tDSJ80foQAS7CZP4=
+```
+
+Then regenerate the `atlas.sum` hashes:
+
+```bash
+make atlas-hash
+```
+
+**N.B.**: If the migration file doesn't have an ".up.sql" extension the
+migration will never be applied. We hope to fix the migration process so
+renaming the file is not necessary, but we haven't found an easy way to do so
+yet.
+
+## Generate a new "enduro_storage" database migration
+
+### Update the "storage" ent schema
+
+Make the required schema changes to the appropriate file in the
+`internal/storage/persistence/ent/schema/` directory, then regenerate the [Ent]
+schema:
+
+```bash
+make gen-ent
+```
+
+### Generate the "storage" migration file
+
+```bash
+go run ./cmd/migrate/ \
+--db="storage" \
+--dsn="mysql://root:root123@tcp(localhost:3306)/enduro_migrate" \
+--path="./internal/storage/persistence/migrations" \
+--name="add_aip_note_column"
+```
+
+See the "enduro" migration generation instructions above for instructions on
+naming the migration.
+
+### Add ".up" to the "storage" migration file extension
+
+Change the migration file name to make the extension ".up.sql":
+
+```bash
+mv \
+internal/storage/persistence/migrations/20250423225046_add_aip_note_column.sql \
+internal/storage/persistence/migrations/20250423225046_add_aip_note_column.up.sql
+```
+
+Also change the file name in the `internal/storage/persistence/migrations/atlas.sum`
+file.
+
+```text
+20250423225046_add_aip_note_column.up.sql h1:Q7c6YSYzIO30VnKzgGdupGih93N81rkmptb6SYDOkao=
+```
+
+Then regenerate the `atlas.sum` hashes:
+
```

Comment:
I'd like to do this in the actual migrate command appending `.up` to the passed name if it's not included already, and just add a note here for migrations created manually altogether.

Reply:
I started trying to add the ".up" to the filename in the migrate command, but it's not trivial - you need to find the migration file name (with the prepended timestamp) then update the name of the file, the `atlas.sum` file and regenerate the file hashes.  I decided to just document the current state as-is, with the intention of improving the migrate command later.

Reply:
I mean adding the `.up` suffix before the migration is generated, in https://github.com/artefactual-sdps/enduro/blob/main/cmd/migrate/main.go#L108:

```go
name, _ := p.GetString("name")
if !strings.HasSuffix(name, ".up") {
	name = fmt.Sprintf("%s.up", name)
}
```

This seems to work for me and it avoids all that manual process for generated migrations.

Reply:
Ah, that is a simple hack that does the job.  I was overthinking the problem I guess. :laughing: 

I've added this change, and updated the documentation appropriately.